libs/api/Verifier.cpp

```diff
@@ -652,8 +585,23 @@ ebpf_api_elf_disassemble_section(
     ebpf_clear_thread_local_storage();
 
     try {
+        std::string section(section_name ? section_name : "");
         auto raw_programs = read_elf(file, section, &verifier_options, platform);
-        raw_program raw_program = raw_programs.back();
+        std::optional<raw_program> found_program;
+        for (auto& program : raw_programs) {
```

Comment:
Consider using std::find_if for more idiomatic code?

Reply:
Done